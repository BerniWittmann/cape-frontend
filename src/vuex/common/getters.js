/* ============
 * Getters
 * ============
 *
 */

const factorRegex = new RegExp('(([a-z]|[0-9]){24})(?=.([a-z]|[0-9]){24})', 'g')

export default {
  graphEdges(state, getters) {
    return [...getters.graphEdgesProcessSituation, ...getters.graphEdgesSituationFactor]
  },

  graphEdgesProcessSituation(state) {
    return state.injectionMapping.injectionMappings.map((injectionMapping) => ({
      start: injectionMapping.processID,
      end: injectionMapping.contextSituation.id,
      type: 'process_situation'
    }))
  },

  graphEdgesSituationFactor(state) {
    const result = []
    state.contextSituation.contextSituations.forEach(situation => {
      const factors = new Set()
      let currentMatch
      while ((currentMatch = factorRegex.exec(situation.rules)) !== null) {
        factors.add(currentMatch[0])
      }

      factors.forEach(factor => {
        result.push({
          start: situation.id,
          end: factor,
          type: 'situation_factor'
        })
      })
    })
    return result
  }
}
