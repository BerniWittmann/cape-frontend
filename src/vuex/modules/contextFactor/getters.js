/* ============
 * Getters for the contextFactor  module
 * ============
 *
 * The getters that are available on the
 * contextFactor  module.
 */

function getRoots(state) {
  return state.contextFactors.filter(cF => !cF.parentID)
}

function getChildren(state, node) {
  return state.contextFactors.filter(cF => cF.parentID === node.id)
}

function constructChildren(state, node) {
  return getChildren(state, node).map(child => ({
    label: child.name,
    contextFactor: child,
    children: constructChildren(state, child)
  }))
}

export default {
  contextFactorsTree(state) {
    return getRoots(state).map(root => ({
      contextFactor: root,
      label: root.name,
      children: constructChildren(state, root)
    }))
  },

  contextFactorGraphNodes(state) {
    return state.contextFactors.map((cF) => ({
      id: cF.id,
      name: cF.name,
      type: 'factor',
      route: {
        name: 'context_factors.edit',
        params: {
          contextFactorID: cF.id
        }
      }
    }))
  }
}
