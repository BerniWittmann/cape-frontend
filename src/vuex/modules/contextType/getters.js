/* ============
 * Getters for the Context Type module
 * ============
 *
 * The getters that are available on the
 * Context Type module.
 */

export default {}
