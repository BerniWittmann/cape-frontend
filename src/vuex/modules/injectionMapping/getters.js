/* ============
 * Getters for the injectionMapping  module
 * ============
 *
 * The getters that are available on the
 * injectionMapping  module.
 */

export default {
  getInjectionMappings: (state) => (processID, extensionAreaID) => {
    return state.injectionMappings.filter(im => {
      return im.processID === processID && im.extensionAreaID === extensionAreaID
    })
  }
}
