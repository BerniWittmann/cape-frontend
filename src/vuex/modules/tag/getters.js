/* ============
 * Getters for the tag module
 * ============
 *
 * The getters that are available on the
 * tag module.
 */

export default {}
