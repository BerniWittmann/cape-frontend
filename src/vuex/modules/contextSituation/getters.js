/* ============
 * Getters for the contextSituation  module
 * ============
 *
 * The getters that are available on the
 * contextSituation  module.
 */

import { getGraphNodes } from '@/vuex/common/helpers'

export default {
  contextSituationGraphNodes(state) {
    return getGraphNodes(state.contextSituations, 'situation', 'context_situations.single', 'contextSituationID')
  }
}
