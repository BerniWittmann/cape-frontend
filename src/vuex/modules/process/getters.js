/* ============
 * Getters for the process module
 * ============
 *
 * The getters that are available on the
 * process module.
 */
import { getGraphNodes, getByTags } from '@/vuex/common/helpers'

export default {
  processGraphNodes(state) {
    return getGraphNodes(state.processes, 'process', 'process.preview', 'processID')
  },

  processesByTags(state, getters, rootState) {
    return getByTags(state.processes, rootState.tag.tags)
  }
}
