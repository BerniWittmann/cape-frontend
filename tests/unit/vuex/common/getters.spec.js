import getters from '@/vuex/common/getters'

describe('Vuex', () => {
  describe('Common', () => {
    describe('Getters', () => {
      it('returns all getters', () => {
        expect(Object.keys(getters)).toMatchSnapshot()
      })

      describe('graphEdges', () => {
        const graphEdges = getters.graphEdges
        let state
        let baseGetters
        beforeEach(() => {
          state = {
            foo: 'bar'
          }
          baseGetters = {
            graphEdgesProcessSituation: [{ id: 1 }, { id: 2 }],
            graphEdgesSituationFactor: [{ id: 3 }, { id: 4 }]
          }
        })

        it('returns an array', () => {
          expect(graphEdges(state, baseGetters)).toEqual(expect.any(Array))
          expect(graphEdges(state, baseGetters)).toEqual([{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }])
        })
      })

      describe('graphEdgesProcessSituation', () => {
        const getter = getters.graphEdgesProcessSituation
        let state
        beforeEach(() => {
          state = {
            injectionMapping: {
              injectionMappings: [{
                id: 'im1',
                processID: 'p1',
                contextSituation: {
                  id: 'cS1'
                }
              }, {
                id: 'im2',
                processID: 'p1',
                contextSituation: {
                  id: 'cS3'
                }
              }, {
                id: 'im3',
                processID: 'p2',
                contextSituation: {
                  id: 'cS3'
                }
              }, {
                id: 'im4',
                processID: 'p1',
                contextSituation: {
                  id: 'cS4'
                }
              }]
            }
          }
        })

        it('returns an array', () => {
          expect(getter(state)).toEqual(expect.any(Array))
        })

        it('returns an empty array if no injectionMappings are available', () => {
          state = {
            injectionMapping: {
              injectionMappings: []
            }
          }
          expect(getter(state)).toEqual([])
        })

        it('returns the edges between processes and situations', () => {
          expect(getter(state)).toEqual([
            { start: 'p1', end: 'cS1', type: 'process_situation' },
            { start: 'p1', end: 'cS3', type: 'process_situation' },
            { start: 'p2', end: 'cS3', type: 'process_situation' },
            { start: 'p1', end: 'cS4', type: 'process_situation' }
          ])
        })
      })

      describe('graphEdgesSituationFactor', () => {
        const getter = getters.graphEdgesSituationFactor
        let state
        beforeEach(() => {
          state = {
            contextSituation: {
              contextSituations: []
            }
          }
        })

        it('returns an array', () => {
          expect(getter(state)).toEqual(expect.any(Array))
        })

        it('returns an empty array if no contextSituations are available', () => {
          state = {
            contextSituation: {
              contextSituations: []
            }
          }
          expect(getter(state)).toEqual([])
        })

        it('returns an empty array if no rules are available', () => {
          state = {
            contextSituation: {
              contextSituations: [{
                id: 'cS1',
                rules: ''
              }, {
                id: 'cS2',
                rules: '000000000000000000000000'
              }, {
                id: 'cS3',
                rules: '12312414'
              }, {
                id: 'cS4',
                rules: 'invalid'
              }]
            }
          }
          expect(getter(state)).toEqual([])
        })
        it('returns the edges between situations and factors', () => {
          state = {
            contextSituation: {
              contextSituations: [{
                id: 'cS1',
                rules: '00000000000000000000000a.ffffffffffffffffffffffff'
              }, {
                id: 'cS2',
                rules: '!00000000000000000000000a.ffffffffffffffffffffffff'
              }, {
                id: 'cS3',
                rules: '00000000000000000000000a.ffffffffffffffffffffffff && !00000000000000000000000b.ffffffffffffffffffffffff'
              }, {
                id: 'cS4',
                rules: '00000000000000000000000c.ffffffffffffffffffffffff && !00000000000000000000000d.ffffffffffffffffffffffff || 00000000000000000000000e.ffffffffffffffffffffffff'
              }]
            }
          }
          expect(getter(state)).toEqual([
            { start: 'cS1', end: '00000000000000000000000a', type: 'situation_factor' },
            { start: 'cS2', end: '00000000000000000000000a', type: 'situation_factor' },
            { start: 'cS3', end: '00000000000000000000000a', type: 'situation_factor' },
            { start: 'cS3', end: '00000000000000000000000b', type: 'situation_factor' },
            { start: 'cS4', end: '00000000000000000000000c', type: 'situation_factor' },
            { start: 'cS4', end: '00000000000000000000000d', type: 'situation_factor' },
            { start: 'cS4', end: '00000000000000000000000e', type: 'situation_factor' }
          ])
        })
      })
    })
  })
})
