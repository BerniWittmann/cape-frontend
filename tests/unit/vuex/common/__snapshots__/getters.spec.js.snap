// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vuex Common Getters returns all getters 1`] = `
Array [
  "graphEdges",
  "graphEdgesProcessSituation",
  "graphEdgesSituationFactor",
]
`;
